### PR TITLE
configure contiv policy from k8s network policy.

### PR DESCRIPTION
**Pull request by rchirakk**
Support k8s network policy in Contiv
documentation in  docs/NetworkPolicy.md

### CODE REVIEW

File: mgmtfn/k8splugin/driver.go [hunk 1]
```diff
@@ -90,6 +92,26 @@ func netdGetNetwork(networkID string) (*mastercfg.CfgNetworkState, error) {
 	return nwCfg, nil
 }
 
+// netdGetEpg is a utility that reads the n/w oper state
+func netdGetEpg(epgName, tenantName string) (*mastercfg.EndpointGroupState, error) {
+	// Get hold of the state driver
```
Comment (gkvijay) [inline mgmtfn/k8splugin/driver.go]: netd prefix is used for functions in netDriver.c. Ideally we should rename and move these to utils

File: netmaster/k8snetwork/networkpolicy.go [hunk 2]
```diff
@@ -47,311 +80,491 @@ func (k8sNet *k8sContext) handleK8sEvents() {
 	}
 }
 
-func (k8sNet *k8sContext) createNetwork(nwName string) error {
-	npLog.Infof("create network %s", nwName)
+func (k8sNet *k8sContext) createEpg(tenantName string, nwName, epgName string) error {
+	policyName := k8sutils.EpgNameToPolicy(epgName)
```
Comment (gkvijay) [inline netmaster/k8snetwork/networkpolicy.go]: k8s policies will end up creating EPGs. We should make sure these are not deleted if the user also creates policies in this EPG

File: netmaster/k8snetwork/networkpolicy.go [hunk 3]
```diff
@@ -16,25 +16,58 @@ import (
...
 func (k8sNet *k8sContext) handleK8sEvents() {
+	// wait for leader election
 	for k8sNet.isLeader() != true {
 		time.Sleep(time.Second * 10)
 	}
 
+	// wait till network creation
+	for func() error {
+		if nwList, err := k8sNet.contivClient.NetworkList(); err == nil {
+			if nwList != nil && len(*nwList) > 0 {
```
Comment (gkvijay) [inline netmaster/k8snetwork/networkpolicy.go]: Shouldn't you wait for group creation as well, to get the configured groups before you start creating epg for k8s np.

File: mgmtfn/k8splugin/driver.go (hunk 1 above)
Comment (rchirakk) [inline mgmtfn/k8splugin/driver.go]: yes, there are a few netdXXX functions already here, that is why I added it here.
Let us move it to utility in another PR
https://github.com/contiv/netplugin/issues/908

File: mgmtfn/k8splugin/driver.go [hunk 4]
```diff
@@ -115,6 +137,84 @@ func epCleanUp(req *epSpec) error {
 	return err2
 }
 
+func createEpgAndDefaultPolicy(tenantName, networkName, epgName string) error {
+	// create epg if it doesn't exists
+	if _, err := netdGetEpg(epgName, tenantName); err == nil {
+		return nil
```
Comment (rchirakk) [inline mgmtfn/k8splugin/driver.go]: createEpgAndDefaultPolicy() created epg and policy,
updated comments.

File: netmaster/k8snetwork/networkpolicy.go (hunk 3 above)
Comment (rchirakk) [inline netmaster/k8snetwork/networkpolicy.go]: no, epg is created if it is not present.
